source/common/util/path-exists.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * BEGIN HEADER
+ *
+ * Contains:        Utility function
+ * CVM-Role:        <none>
+ * Maintainer:      Patrik Andersson
+ * License:         GNU GPL v3
+ *
+ * Description:     This file contains a utility function to check if a path exists.
+ *
+ * END HEADER
+ */
+
+import { type PathLike, existsSync } from 'fs'
+
+/**
+ * Check if a path exists.
+ *
+ * @param   {PathLike}  p  path to be tested
+ * @return  {boolean}      true if the path exists otherwise false
+ */
+export default function pathExists (p: PathLike): boolean {
+  return existsSync(p)
```

Comment:
Do not use any synchronous methods for file access. Please take inspiration from other files; usually we import `fs.promises` and `await` results.

Reply:
It says that fs.exists() is deprecated but not existsSync. I also checked the is-file and is-dir code and they are using lstatSync which is synchronous code. But I'm considering to upload a changeset with an async pathExists

```typescript
export default async function pathExists (p: PathLike): Promise<boolean> {
  let stat, stat2
  try {
    stat = await promises.lstat(p)
  } catch (err) {
    return false
  }
  return stat !== undefined
}
```

Is that what you've got in mind?

Reply:
the is-file and is-dir is not just a search and replace. The will be awaits ending up in constructors of object so there are some thoughts needed there to get it right.